cis194/week7/tien/src/JoinList.hs

```diff
@@ -1,4 +1,6 @@
 {-# OPTIONS_GHC -Wall #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeSynonymInstances #-}
```

Comment:
FYI `TypeSynonymInstances` is implicitly applied with the application of `FlexibleInstances`, so you don't need it here.